ci-operator/step-registry/ipi/conf/aws/oidc-creds-provision/ipi-conf-aws-oidc-creds-provision-commands.sh

```diff
@@ -4,12 +4,173 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+trap 'rm -f /tmp/aws_cred_output' EXIT TERM INT
+
 MPREFIX="${SHARED_DIR}/manifest"
 TPREFIX="${SHARED_DIR}/tls"
-infra_name=${NAMESPACE}-${UNIQUE_HASH}
-export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+CLUSTER_NAME=${NAMESPACE}-${UNIQUE_HASH}
 REGION="${LEASED_RESOURCE}"
 
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+# -------------------------------------------
+# config AWS credential for ccoctl tool
+# -------------------------------------------
+function run_command() {
+    local cmd="$1"
+    echo "Running Command: ${cmd}"
+    eval "${cmd}"
+}
+
+function aws_create_policy()
+{
+    local aws_region=$1
+    local policy_name=$2
+    local policy_doc=$3
+    local output_json="$4"
+
+    cmd="aws --region $aws_region iam create-policy --policy-name ${policy_name} --policy-document '${policy_doc}' > '${output_json}'"
+    run_command "${cmd}" || return 1
+    return 0
+}
+
+function aws_create_user()
+{
+    local aws_region=$1
+    local user_name=$2
+    local policy_arn=$3
+    local user_output=$4
+    local access_key_output=$5
+    
+    # create user
+    cmd="aws --region ${aws_region} iam create-user --user-name ${user_name} > '${user_output}'"
+    run_command "${cmd}" || return 1
+
+    # attach policy
+    cmd="aws --region ${aws_region} iam attach-user-policy --user-name ${user_name} --policy-arn '${policy_arn}'"
+    run_command "${cmd}" || return 1
+
+    # create access key
+    cmd="aws --region ${aws_region} iam create-access-key --user-name ${user_name} > '${access_key_output}'"
+    run_command "${cmd}" || return 1
+
+    return 0
+}
+
+
+if [[ ${AWS_INSTALL_USE_MINIMAL_PERMISSIONS} == "yes" ]]; then
```

Comment:
Actually personally I  like your origin idea - "provision mini-perm user for ccoctl in provision-iam-user" step in https://github.com/openshift/release/pull/58622.

Reply:
@jianlinliu  it looks like we need a separate step for creating user for ccoctl, which will be executed before before `chain: aws-provision-iam-user-minimal-permission`:
```
Creatingc IAM user with minimal permission for ccoctl
...
environment: line 25: aws: command not found 
```

Reply:
how about:
```
chain:
  as: ipi-conf-aws-oidc-creds-provision
  steps:
    - ref: aws-provision-iam-user-ccoctl
    - ref: ipi-conf-aws-oidc-creds-provision
```

Reply:
Then `ipi-conf-aws-oidc-creds-provision` must be in the between `- chain: aws-provision-iam-user-minimal-permission` and `ipi-install`:
- ...
- chain: aws-provision-iam-user-minimal-permission
- ref: ipi-conf-aws-oidc-creds-provision
- [ipi-install]
- ...

Reply:
Right, sounds like only QE have the install workflow to cover mini-perm + oidc, so personally I think it is okay that we can have some adjustment on the step order in our provision chain.

Reply:
> sounds like only QE have the install workflow to cover mini-perm + oidc

Right.
Let me update

Reply:
Updated.